File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    dist = :math.sqrt(x*x+y*y)
    cond do
      dist > 10 -> 0
      dist > 5 -> 1
      dist > 1 -> 5
      true -> 10
    end
  end
end

File: solutions/go/darts/1/darts.go
package darts

import "math"

func Score(x, y float64) int {
	radius := math.Sqrt(x*x+y*y)
    switch {
        case radius <= 1: return 10
        case radius <= 5: return 5
        case radius <= 10: return 1
        default: return 0
    }
}
